lua/typescript-tools/protocol/text_document/code_action/init.lua

```diff
@@ -86,6 +94,20 @@ function M.handler(request, response, params, ctx)
     })
   end
 
+  for _, cmd in ipairs(plugin_config.expose_as_code_action) do
```

Comment:
So here we are putting it **always** into code actions, but as far as I know VScode (or vstls) has more sophisticated algorithm to do that (example will be for `fix_all` but the same applies to other types):
1. If there are no code actions it does not put it in the code actions response
2. If there is single fixable by `fix_all` diagnostic and cursor is on the diagnostic it only puts this single code action (without `fix_all` action)
3.  If cursor is on the same diagnostic and in the file there is **more than one** fixable diagnostic, only then there is code action `fix_all`
That being said i think it's more understandable approach, because in our case, in most cases, e.g. `fix_all` code action would do nothing at all
Just leaving this here in case you would like to do the same here